distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso

```diff
@@ -174,6 +174,22 @@ type DB_Table
         col = self.get selector if_missing=Nothing
         if Nothing == col then if_missing else col.get index if_missing
 
+    ## ALIAS row
+       GROUP Standard.Base.Selections
+       ICON select_row
+       Gets a row from the table.
+
+       Arguments:
+       - index: The index of the row to get within the table.
+       - if_missing: The value to use if the selector isn't present.
+    @index (t-> Widget.Numeric_Input minimum=0 maximum=t.row_count-1)
+    get_row : Integer -> Any -> Any
+    get_row self index:Integer=0 ~if_missing=Nothing =
+        if index < -self.row_count || index >= self.row_count then if_missing else
```

Comment:
Unfortunately, `row_count` is **not** cached. So calling it twice means we call the query twice... That is highly undesirable.

Please assign the `self.row_count` result to a variable and use that.

We could technically add a `Ref` field to `Context` and cache the `row_count` there if you want. But the problem is it can actually get updated if someone else updates the table - so it's not really robust. I think we just need to avoid recomputing it too much.

Reply:
I'm wondering if the DB variant should attach a warning.

The DB is not guaranteeing order of rows. There is no guarantee that even if data is unchanged, that subsequent runs of the query will return the rows in the same order.

So technically `t.get_row 0` and `t.get_row 1` could give us the same row if the results get reordered in the meantime.

Of course, in practice this will be probably quite unlikely to happen. But there are no guarantees - so IMO we should at least mention that in the doc.

Reply:
@radeusgd think better implementation now